### PR TITLE
Jocelyn Gonzalez -- Carets

### PR DESCRIPTION
**Pull request by jocegonz**
# Grocery Store
Congratulations! You're submitting your assignment.

## Comprehension Questions

|  Question 	|   Response	|
|---	|---	|
|   Why is it useful to put classes inside modules? 	| When we use modules as a namespace, we're able to bundle similar objects together. By belonging to the same module, we're able to inherit code - which makes it much easier when the code is similar.  	|
|   What is accomplished with `raise ArgumentError`? 	|   Raise ArgumentError should be alerted when a parameter is outside its bounds. It's great because if it's called, we have a note that tells us where the issue is.	|
|   Why do you think we made the `.all` & `.find` methods class methods?  Why not instance methods?	|   .all and .find act on the classes that are calling them. In these cases, they weren't instantiated by an outside variable because the information it needed was located inside the CSV files.	|
|   Why does it make sense to use inheritance for the online order?  |  Online order had a lot of the same behavior as order. In the case of self.find(), it needed no refactoring at all. In the case of .total, it only needed a slight adjustment to add shipping. |
|   Did the presence of automated tests change the way you thought about the problem? How? | I think I was able to see the bigger picture a bit better. Though I was testing the code for edge cases, it forced me to be more directed than I usually am. It made me consider more uniformity in my code. Especially in terms of inheritance, I had to consider if changing the structure of my code changed the way it would function/error out.

### CODE REVIEW

**Comment by tildeee** [comment]
# Grocery Store
## What We're Looking For

|  Feature  |   Feedback|
|---	|---	|
|  **Baseline** |   |
|   Answered comprehension questions	|  x  |
|   Used Git Regularly	|  x  |
|  **Wave 1** |   |
|   All provided tests pass | x |
|   Using the appropriate `attr_` for instance variables |  x  |
|  **Wave 2** |   |
|   All stubbed tests are implemented fully and pass |  |
|   Appropriately parses the product data from CSV file in `Order.all` | x |
|   Used `CSV` library only in `Order.all` (not in `Order.find`) |  x  |
|   Used `Order.all` to get account list in `Order.find`	|  x  |
|  **Wave 3** |   |
|   All stubbed tests are implemented fully and pass |  |
|   Used inheritance in the initialize for online order |  x  |
|   Used inheritance for the total method in online order | x |
|   Appropriately created Customer class and class methods `all` & `find` | x |
|   **Additional Notes**	|   	|

File: lib/online_order.rb [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        end
+      customer_id = line.shift
+      status = line.join() #status should be converted to symbol in initialize
+      online_order_array << Grocery::OnlineOrder.new(id, products, customer_id, status)
+    end
+    return online_order_array
+  end
+
+  def self.find(id)
+    super
```
Comment (tildeee) [inline lib/online_order.rb:49]: not appropriate: this calls the exact implementation of the `Order#find` method

File: specs/online_order_spec.rb [hunk 2]
```diff
@@ -3,50 +3,108 @@
...
+      order_plus_shipping = order.total
+      puts "Total_3_Products: #{order_plus_shipping}"
+
     end
 
     it "Doesn't add a shipping fee if there are no products" do
       # TODO: Your test code here!
+      order = Grocery::OnlineOrder.new(7, {}, 24, :complete)
+      order_plus_shipping = order.total
+      puts "Total_0_Products: #{order_plus_shipping}"
```
Comment (tildeee) [inline specs/online_order_spec.rb:91]: Don't forget to remove all `puts` statements before turning in. `puts` does not create a valid test.

**Comment by tildeee** [comment]
Your tests were mostly incomplete or insufficient for Wave 2 and Wave 3-- it is not testing your code thoroughly enough to check if some output is the object type you are looking for. In tests, we are hoping to check the validity and integrity of the data returned, so we expect you to test against the content of those values as well. Let me know if you have questions.
